### PR TITLE
Weather App Javascript

### PR DESCRIPTION
**Pull request by gitnyasha**
- weather information of each searched city
- toggle-button to change Fahrenheit and celsius

### CODE REVIEW

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 

Hi @gitnyasha 👋 

You have done a good job thus far! 🙁 

Just one required feature missing. 😅 

#### Required changes
- you are missing the feature that changes a feature of the app base on type of weather city returns. See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from Odin.
- Your temperature toggle also does not always work. When I search for some cities it does not toggle the temperature please address this bug.

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
Changes Required :muscle: 
Hi there @gitnyasha 
Great job on this project, but we still have some changes to work on.

- [ ] The application logic should be separated from the DOM manipulations tasks. [Print](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) of the requirement.

- [ ] There is a bug in your app. The change between Celsius and Fahrenheit does not always happen as expected. Sometimes it works, sometimes it doesn't
bug

- [ ] As it said in the requirements, the change between Celsius and Fahrenheit should trigger another request in the API with the desired scale.
[Print](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/requests_requirement.png) of the requirement

Please kindly submit for code review again after making these changes.
Happy Coding :computer: 

**Sergio Torres**

File: src/weather.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const response = await fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${this.city}&appid=${this.apiKey}`,
+    );
+    const responseData = await response.json();
+
+    return responseData;
+  }
+
+  changeLocation(city) {
+    this.city = city;
```
Comment (bellom) [inline src/weather.js:17]: According to the previous review, @Torres-ssf mentioned calling different `API`

kindly check the screenshot from the pathwright below;
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import Dom from './views';
+
+const weather = new Weather('Harare');
+const dom = new Dom();
+
+function getWeather() {
+  weather.getWeather().then((results) => {
+    dom.show(results);
+  });
+}
```
Comment (bellom) [inline src/index.js:11]: According to the previous review, @Torres-ssf mentioned that the `toggle button` is not working properly, kindly look into this, still not working.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @gitnyasha   :clap: 

**For some reason I didn't see your recent changes**

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by shloch — APPROVED** [review]
### Good to go.....you get a thumbs up

weather

**Pull request merged**